code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -517,27 +519,47 @@
 		ui.set_auto_update(1)
 
 /obj/item/weapon/rig/update_icon(var/update_mob_icon)
-
 	overlays.Cut()
+
 	if(!mob_icon || update_mob_icon)
-		var/species_icon = 'icons/mob/rig_back.dmi'
-		// Since setting mob_icon will override the species checks in
-		// update_inv_wear_suit(), handle species checks here.
-		mob_icon = image("icon" = species_icon, "icon_state" = icon_state)
+		mob_icon = image("icon" = onmob_effect_icon, "icon_state" = icon_state)
 
-	if(installed_modules.len)
-		for(var/obj/item/rig_module/module in installed_modules)
-			if(module.suit_overlay)
-				chest.overlays += image("icon" = 'icons/mob/rig_modules.dmi', "icon_state" = module.suit_overlay, "dir" = SOUTH)
+	add_item_overlay(chest)
 
 	if(wearer)
 		wearer.update_inv_shoes()
 		wearer.update_inv_gloves()
 		wearer.update_inv_head()
+		wearer.update_inv_wear_mask()
 		wearer.update_inv_wear_suit()
+		wearer.update_inv_w_uniform()
 		wearer.update_inv_back()
```

Comment:
Why so much overlays should be updated?

Reply:
This is left after first iteration, will correct